packages/web/src/graphql/withCell.tsx

```diff
@@ -1,5 +1,24 @@
 import React from 'react'
 import { Query } from '@apollo/client/react/components/Query'
+import { DocumentNode } from 'graphql'
+import {
+  BaseQueryOptions,
+  OperationVariables,
+  QueryResult,
+} from '@apollo/client'
+
+type DataObject = { [key: string]: unknown }
+
+type QueryResultAlias = QueryResult<any, Record<string, any>>
+
+type CellFailureStateComponent = Omit<QueryResultAlias, 'data' | 'loading'>
+type CellLoadingEmptyStateComponent = Omit<
+  QueryResultAlias,
+  'error' | 'loading' | 'data'
+>
+type CellSuccessStateComponent =
+  | Omit<QueryResultAlias, 'error' | 'loading' | 'data'>
+  | DataObject
```

Comment:
Do you think it would be useful to export these types?

Reply:
I can't really think of a case for using them right now, but I also don't see any harm in exporting them.